### DISCUSSION TITLE
Dodona 5 redesign

### DISCUSSION THREAD

**Discussion by bmesuere** [dodona-edu/feedback-and-support#5] [General]
This topic is used to discuss the individual ideas implemented in #2760.

Please use threaded comments to give feedback.

**Update**: Most of these changes were implemented. I'll leave this thread open for now to gather feedback after version 5.0 is released.

**Comment by bmesuere** [comment]
### Increase the border radius for cards

**Motivation**
A more playful appearance.

**Changes**
Change border-radius from 6px to 12px.

**Status**
Implemented in #3024 

**Before**
![image](https://user-images.githubusercontent.com/481872/119271961-287b6a80-bc04-11eb-9651-bb18b51b76a7.png)

**After**
![image](https://user-images.githubusercontent.com/481872/119272067-8c059800-bc04-11eb-9949-4db975b1374c.png)

**Comment by bmesuere** [comment]
### Don't uppercase button text

**Motivation**
Less aggressive style and a bit more compact for longer labels.

**Changes**
Remove the text-transform uppercase from buttons.

**Status**
Implemented in #3024 

**Potential future changes**
* Bring the padding in line with the material design spec (should be 16 instead of 8px for the primary button)
* Try to remove the manual button style overrides and try to replace them with variables

**Before**
![image](https://user-images.githubusercontent.com/481872/119272198-1fd76400-bc05-11eb-8378-7880eac42de9.png)

**After**
![image](https://user-images.githubusercontent.com/481872/119272275-7775cf80-bc05-11eb-8a1f-87696d25e2cf.png)

**Reply by niknetniko** [reply]
I'm not fully convinced on this, as the secondary buttons loose a lot of their distinction from normal links (the only difference as far as I can tell is that the text is bold, while links are generally not).

**Reply by bmesuere** [reply]
I'm not sure this is a big issue because in a lot of cases, they actually are links and not "actions". In combination with their specific placement (and :hover), it's still clear they are buttons. For example on the home page it seems quite natural to me:
<details>
<summary>Screenshot</summary>

![image](https://user-images.githubusercontent.com/481872/119303461-0adeed00-bc66-11eb-839c-7d39bd305150.png)

</details>

**Comment by bmesuere** [comment]
### ✅  Wider drawer

**Motivation**
The material design spec suggests using a width of 256px, we used 240.

**Changes**
Increase the drawer width from 240px to 256px.

**Potential future changes**
* Disable line breaks for longer menu items and add an ellipsis overflow.

**Status**
Implemented in #2887

<table>
<tr><td><strong>Before</strong></td><td><strong>After</strong></td></th>
<tr><td>

![image](https://user-images.githubusercontent.com/481872/119272628-2ff04300-bc07-11eb-862d-d324253196c9.png)
</td><td>

![image](https://user-images.githubusercontent.com/481872/119272636-37175100-bc07-11eb-9798-b81213834ec0.png)</td>
</tr></table>

**Reply by bmesuere** [reply]
This was released as part of Dodona 4.7 🚀

**Comment by bmesuere** [comment]
### Rework tokens

**Motivation**
The current tokens were very compact and at the same time quite heavy because of the dark background color and border. This change lightens things up.

**Changes**
Set the background color to the *-200 value of each color, set the text color to black with 75% opacity, remove the border and bring the padding and margin in line with the spec, capitalize the first letter.

**Status**
Implemented in #3020 

**Before**
![image](https://user-images.githubusercontent.com/481872/119272928-ca04bb00-bc08-11eb-95f1-f8c43141cb7b.png)

**After**
![image](https://user-images.githubusercontent.com/481872/119272985-194aeb80-bc09-11eb-9812-e70de4e3dd8c.png)

**Reply by niknetniko** [reply]
I'm not sure if the background colours aren't too dark (resulting in not enough contrast). Sometimes when I look at it it seems fine, but other times it seems to lack contrast (so unsure if this is an actionable comment).

(but they are an improvement over the existing ones, whether we change the contrast or not)

**Reply by bmesuere** [reply]
Believe it or not, but the original tokens have an accessibility warning because, according to chrome, their contrast isn't high enough (3.36 for the green token) while the new ones pass the Aa level (7.12 for the green one).

**Comment by bmesuere** [comment]
### ✅ Extra submission feedback

**Motivation**
Correctly solving an exercise should be a rewarding experience. With this change, after a submission is judged, the FAB temporarily changes color and icon to indicate the result of the submission.

**Changes**
For the duration of 4 seconds, the color and icon of the FAB is changed after judging is completed. If the submission is incorrect, a sad emoji is shown on a red background. If the submission is correct, a positive emoji is shown on a green background. The positive emoji's are picked at random from a list of 10 with a strong bias towards the first elements in the list (`emojis[Math.floor(Math.pow(Math.random(), 3) * emojis.length)]`). This will make some results rare than others.

Currently, the list of positive icons is: `["check-bold", "thumb-up-outline", "emoticon-happy-outline", "emoticon-excited-outline", "emoticon-cool-outline", "emoticon-kiss-outline", "robot-outline", "unicorn-variant", "cow", "emoticon-poop"]`.

**Status**
Implemented in #2889

**Positive result**

https://user-images.githubusercontent.com/481872/119645604-7ff52280-be1e-11eb-9211-e87101178b1d.mov

**Negative result**

https://user-images.githubusercontent.com/481872/119646092-0f9ad100-be1f-11eb-9d9f-ff2baf9de08e.mov

**Reply by pdawyndt** [reply]
Seems like the icon is not 100% centered in the button.

**Reply by pdawyndt** [reply]
This one definitely will be an eye-catcher!

**Reply by bmesuere** [reply]
This was released as part of Dodona 4.7 🚀

**Comment by bmesuere** [comment]
### Tweak home page colors

**Motivation**
We currently use saturated colors for the home page cards in combination with white text. The potential mix of prominent colors diverts attention from the actual content (and the contrast didn't always pass the accessibility thresholds) . This update switches to more muted, desaturated pastel colors with a darker font color. These are the same colors that are used for the token rework.

**Changes**
Set the background color to the *-200 value of each color, set the text color to black with 75% opacity for the primary text and 60% opacity for the secondary text.

**Status**
Implemented in #3024 

**Before**
![image](https://user-images.githubusercontent.com/481872/124929104-828f8e00-e000-11eb-856a-16996ec4caef.png)

**After**
![image](https://user-images.githubusercontent.com/481872/124928910-5247ef80-e000-11eb-8a53-08e85fd3dba4.png)

**Reply by bmesuere** [reply]
An issue here is the multitude of colors on the home page. This is also an issue with the current design and is amplified with the the home page action cards idea. We can only use a single color, but this might be a bit boring. An alternative is to use a single color for each academic year. This will group the colors for users with a lot of cards.

**Reply by BTWS2** [reply]
Maybe Dodona can let the teachers change the color in the settings: https://dodona.ugent.be/en/courses/???/edit/ (first mentioned in #2240)? I prefer the saturated colors, because they match with the Dodona blue. My suggestion would be to use the same color for the course and in the overview to be consistent. This helps to tell the difference between different courses and the same course of a different academic year.

**Comment by bmesuere** [comment]
### Home page action cards

**Motivation**
The homepage currently is quite static and could do more to surface relevant information. Especially for users that are only subscribed to a single course, the home page is relatively empty. This suggestion adds a row of "action cards" to the top of the home page.

**Changes**
- The leftmost card is "jump back in" and is shown in blue. This shows a recent exercise that wasn't yet correctly solved. If there are no such recent exercises, the card could lead to a recent incomplete series. Clicking on the card brings the user to the respective exercise, potentially with their last submission already loaded in the editor.
- Next, a number of relevant "deadline" cards are shown. These cards are sourced from all courses a student is subscribed to. The color of the card depends if the series status. It is green if the user successfully solved all exercises and red otherwise. For teachers, it could indicate how many students already completed the series.
- A maximum of 4 cards is shown.

**Todo**
- Include a marker/icon (bottom right?) to indicate the cards are clickable.
- Include a progress bar and more status information in the deadline cards. We do have to make sure all of the card types have the same height.

**Open questions**
- Should we always show these cards, or only on desktop screens?
- Are there other cards/actions we could show here?
- Can this replace part of the right sidebar?
- I'm not sure about the style of the action cards. An alternative could be to have white cards (with or without drop shadow) with a colored (more saturated) border instead.

**Status**
Still exploring options with static mockup

**Current mockup**
![image](https://user-images.githubusercontent.com/481872/124972566-d44d0e00-e02a-11eb-90b1-deb6d4e90018.png)

**Reply by bmesuere** [reply]
This was moved to a separate issue: #3033
